app/Program.cs

```diff
@@ -25,20 +44,127 @@ static void Main(string[] args)
+                                forcedFallbackFormat = 4,
+                                downscaleFallback = 0,
+                                width = 1024,
+                                height = 1024,
+                                completeImageSize = customSongsCover.Length,
+                                textureFormat = 34,
+                                mipCount = 11,
+                                isReadable = false,
+                                streamingMips = false,
+                                streamingMipsPriority = 0,
+                                imageCount = 1,
+                                textureDimension = 2,
+                                filterMode = 2,
+                                mipBias = -1f,
+                                anisotropic = 0,
+                                wrapU = 1,
+                                wrapV = 1,
+                                wrapW = 0,
+                                lightmapFormat = 6,
+                                colorSpace = 1,
+                                imageData = customSongsCover,
+                                offset = 0,
+                                size = 0,
+                                path = ""
+                            };
+
+                            Console.WriteLine($"Replacing Texture at PathID: {45} with new Texture2D from file: {args[i + 1]}");
+
+                        } catch (FileNotFoundException)
+                        {
+                            Console.WriteLine($"[ERROR] Custom cover file does not exist: {args[i+1]}");
+                        }
+                        i++;
+                        continue;
+                    }
                     Utils.FindLevels(args[i], levelFolder => {
                         try {
                             JsonLevel level = JsonLevel.LoadFromFolder(levelFolder);
                             string levelID = level.GenerateBasicLevelID();
-                            if(existingLevels.Contains(levelID)) {
-                                Console.WriteLine($"Present: {level._songName}");
+                            var apkTxn = new Apk.Transaction();
+
+                            if (existingLevels.Contains(levelID)) {
+                                if (removeSongs)
+                                {
+                                    // Currently does not handle transactions (it half-supports them)
+                                    Console.WriteLine($"Removing: {level._songName}");
+                                    ulong levelPid = assets.GetLevelMatching(levelID).RemoveFromAssets(assets, apkTxn);
+                                    // We also don't _need_ to remove this from the existingLevels, but we probably _should_
+                                    existingLevels.Remove(levelID);
+                                    
+                                    extrasCollection.levels.RemoveAll(ptr => ptr.pathID == levelPid);

```

Comment:
It's hard for me to reason through whether this path ID is still correct after all the shifting. I'm sure you've made it so that it is, but it would be easier to just remove it from the level collection before deleting the assets and then the correctness would be obvious.